test/common/network/connection_impl_test.cc

```diff
@@ -334,6 +334,11 @@ TEST_P(ConnectionImplTest, SocketOptions) {
       }));
 
   dispatcher_->run(Event::Dispatcher::RunType::Block);
+
+  // Assert that upstream connection gets the socket options
+  ASSERT(upstream_connection_ != nullptr);
+  ASSERT(upstream_connection_->socketOptions() != nullptr);
```

Comment:
nit: these 2 ASSERTS will crash in an obvious way on the next line so I would remove, but nbd.